cmake/macros/macro_define_interface_target.cmake

```diff
@@ -142,16 +142,12 @@ function(define_interface_target _feature)
 
     if(${_feature}_SPLIT_CONFIGURATION)
       # Future FIXME: change to block(PARENT_SCOPE)/endblock() (CMake 3.25)
-      list(APPEND DEAL_II_TARGETS_${_build} "${_interface_target}")
+      list(APPEND DEAL_II_TARGETS_${_build} ${_interface_target})
       set(DEAL_II_TARGETS_${_build} ${DEAL_II_TARGETS_${_build}} PARENT_SCOPE)
-      list(APPEND DEAL_II_LIBRARIES_${_build} "${_libraries}")
-      set(DEAL_II_LIBRARIES_${build} ${DEAL_II_LIBRARIES_${build}} PARENT_SCOPE)
     else()
       # Future FIXME: change to block(PARENT_SCOPE)/endblock() (CMake 3.25)
-      list(APPEND DEAL_II_TARGETS "${_interface_target}")
+      list(APPEND DEAL_II_TARGETS ${_interface_target})
       set(DEAL_II_TARGETS ${DEAL_II_TARGETS} PARENT_SCOPE)
-      list(APPEND DEAL_II_LIBRARIES "${_libraries}")
-      set(DEAL_II_LIBRARIES ${DEAL_II_LIBRARIES} PARENT_SCOPE)
```

Comment:
Why are you not setting `DEAL_II_LIBRARIES` here anymore?

Reply:
Because `DEAL_II_LIBRARIES` should ideally be empty and everything should be associated with `interface_*` targets that are recorded in the `DEAL_II_TARGETS` variable.

If we set `DEAL_II_LIBRARIES` in this context then we will link against everything twice.

This reminds me that I need to sit down and update our documentation.